src/components/UploadButton/UploadButton.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {
+  Dispatch,
+  HTMLProps,
+} from 'react';
+import './UploadButton.css';
+
+interface PropTypes extends Omit<
+  HTMLProps<HTMLInputElement>,
+  "id" | "name" | "type" | "onChange"
+> {
+  // Button contents
+  label?: string,
+
+  // This type comes from a built-in React method.
+  // Essentially, its typed to the "setState" function.
+  setFile: Dispatch<File | null>
+};
+
+// Primary UI component for user interaction
+const UploadButton = ({
+  label="Upload a File...",
+  setFile,
+  ...props
+}: PropTypes) => {
+  // Set the state of the file.
+  return (
+    <div>
+      <label htmlFor="file-uploader">
+        {label}
+      </label>
+
+      <input
+        id="file-uploader"
```

Comment:
You are supposed to have unique ids in an html document so I would avoid setting it unless you need it (and I do not think you need it). Name is more flexibile but if you don't need it, no reason to provide it either I think.

Reply:
With https://github.com/AsyncRacing/AsyncRacing/pull/5/commits/1fb6c86b0807bed31fff6914e135287f890d2ac7:
This is resolved as the label / id is no longer in the component